src/globus_sdk/services/auth/response/oauth.py

```diff
@@ -191,7 +191,8 @@ def decode_id_token(
             step. If ``"leeway"`` is included, it will be passed as the ``leeway``
             parameter, and all other values are passed as ``options``.
         """
-        logger.info('Decoding ID Token "%s"', self["id_token"])
+        id_token = self["id_token"]
+        logger.info('Decoding ID Token "%s"', id_token)
```

Comment:
This is tangential do your actual change but flags me as one of those times where the sdk is louder than is should be.
If out-of-scope, you can discard this without comment.

```suggestion
        logger.debug('Decoding ID Token "%s"', id_token)
```

Reply:
Oh, yeah, this can be made into debug. In general the SDK is quite chatty in its logging since the expectation is that library logging will be turned off and that if someone goes out of their way to turn it on, they want _a lot_ of detail on what's happening.